library/insights_config.py

```diff
@@ -0,0 +1,49 @@
+ANSIBLE_METADATA = {
+     'metadata_version': '1.1',
+     'status': ['preview'],
+     'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: insights_config
+short_description: This module handles initial configuration of the insights client on install
+description:
+  - Supply values for various configuration options that you would like to use. On install
+  this module will add those values to the insights-client.conf file prior to registering.
+version_added: "3.0"
+options:
+  insights_name:
+    description:
+    - For now this is just 'insights-client', but that could change in the future if the
+    product name changes.
+    required: true
+  username:
+    description:
+    - Insights basic auth username. If defined this will change, set, or remove the username
+    in the insights configuration. To remove a username set this value to an empty string.
+    required: false
+  password:
+    description:
+    - Insights basic auth password. If defined this will change, set, or remove the password
+    in the insights configuration. To remove a password set this value to an empty string.
+    required: false
+  auto_config:
```

Comment:
Does `auto_config` conflict with when `username` and `password` are provided?  Meaning, can the caller of this module provide one or the other?

Reply:
From what I understand if auto_config is set to true then username/pass/authmethod will be ignored and it will attempt to authenticate via CERT. I'll talk with @gravitypriest on how best to handle that.

Reply:
Cool thanks, yeah that'd be good to ensure we can handle those two registration scenarios, 1) username+pass without a RH subscription, 2) RH subscription.  I don't know if there is a 3) username+pass *with* a RH subscription.